web-api-shared/src/elastic.rs

```diff
@@ -415,33 +420,51 @@ impl Client {
             .await
     }
 
-    pub async fn search_request<I>(&self, mut body: JsonObject) -> Result<HashMap<I, f32>, Error>
+    pub async fn search_request<Id>(&self, mut body: JsonObject) -> Result<ScoreMap<Id>, Error>
     where
-        I: DeserializeOwned + Eq + Hash,
+        Id: FromStr + Eq + Hash,
     {
         if body.get("size") == Some(&json!(0)) {
             return Ok(HashMap::new());
         }
         body.insert("_source".into(), json!(false));
         body.insert("track_total_hits".into(), json!(false));
-        self.query_with_json::<_, SearchResponse<I>>(
-            Method::POST,
-            self.create_url(["_search"], None),
-            Some(body),
-        )
-        .await
-        .map(|response| {
-            response
-                .map(|response| {
-                    response
-                        .hits
-                        .hits
-                        .into_iter()
-                        .map(|hit| (hit.id, hit.score))
-                        .collect()
-                })
-                .unwrap_or_default()
-        })
+        let response = self
+            .query_with_json::<_, SearchResponse<String>>(
+                Method::POST,
+                self.create_url(["_search"], None),
+                Some(body),
+            )
+            .await?;
+
+        let mut map = HashMap::<_, (f32, HashSet<usize>)>::default();
```

Comment:
:pushpin:  for now we track only which splits we found and with a dev option for tests return that but we probably need/want something better (see PR description)